docs/basics.md

```diff
@@ -0,0 +1,160 @@
+**_These docs are for Marionette 3 which is still in pre-release. Some parts may
+not be accurate or up-to-date_**
+
+# Common Marionette Concepts
+
+This document covers the basic usage patterns and concepts across Marionette.
+This includes things like calling conventions, setting attributes, common option
+patterns etc.
+
+## Documentation Index
+
+* [Class-based Inheritance](#class-based-inheritance)
+  * [Value Attributes](#value-attributes)
+  * [Functions Returning Values](#functions-returning-values)
+* [Setting Options](#setting-options)
+  * [The `getOption` Method](#the-getoption-method)
+
+## Class-based Inheritance
+
+Backbone and Marionette utilize the `_.extend` function to simulate class-based
+inheritance. All built-in classes, such as `Marionette.View`,
+`Marionette.Object` and everything that extend these provide an `extend` method
+for just this purpose.
+
+In the below example, we create a new pseudo-class called `MyView`:
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var MyView = Mn.View.extend({});
+```
+
+You can now create instances of `MyView` with JavaScript's `new` keyword:
+
+```javascript
+var view = new MyView();
+```
+
+### Value Attributes
+
+When we extend classes, we can set a form of class attributes by setting new
+values to be a specific value by setting it when we extend our class:
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var MyView = Mn.View.extend({
+  className: 'bg-success',
+
+  template: '#template-identifier',
+
+  regions: {
+    myRegion: '.my-region'
+  },
+
+  modelEvents: {
+    change: 'removeBackground'
+  },
+
+  removeBackground: function() {
+    this.$el.removeClass('bg-success');
+  }
+});
+```
+
+When we instantiate `MyView`, each instance will be given a `.bg-success` class
+with a `myRegion` region created on the `.my-region` element.
+
+### Functions Returning Values
+
+In almost every instance where we can set a value, we can also assign a function
+to figure out the value at runtime. In this case, Marionette will run the
+function on instantiating and use the returned value:
```

Comment:
on instantiation

also might mention something like "with the context of the instance"